pkg/apiserver/registry/controlplane/egressgroup/rest_test.go

```diff
@@ -78,3 +135,62 @@ func TestRESTList(t *testing.T) {
 		})
 	}
 }
+
+func TestRESTWatch(t *testing.T) {
+	tests := []struct {
+		name           string
+		egressGroups   []*types.EgressGroup
+		labelSelector  labels.Selector
+		expectedEvents []watch.Event
+	}{
+		{
+			name: "label selector selecting nothing",
+			egressGroups: []*types.EgressGroup{
+				{
+					Name: "foo",
+				},
+			},
+			labelSelector:  labels.Nothing(),
+			expectedEvents: nil,
+		},
+		{
+			name: "label selector selecting everything",
+			egressGroups: []*types.EgressGroup{
+				{
+					Name: "foo",
+				},
+			},
+			labelSelector: labels.Everything(),
+			expectedEvents: []watch.Event{
+				{
+					Type: watch.Added,
+					Object: &controlplane.EgressGroup{
+						ObjectMeta: v1.ObjectMeta{
+							Name: "foo",
+						},
+					},
+				},
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			storage := store.NewEgressGroupStore()
+			for _, obj := range tt.egressGroups {
+				storage.Create(obj)
+			}
+			r := NewREST(storage)
+			watcher, err := r.Watch(context.TODO(), &internalversion.ListOptions{LabelSelector: tt.labelSelector})
+			assert.NoError(t, err)
+			defer watcher.Stop()
+			for _, expectedObj := range tt.expectedEvents {
```

Comment:
for the `"label selector selecting nothing"` test case, I don't think we are actually testing anything? at least it doesn't seem that we are checking that no events are received on the channel.

Reply:
Thanks for catching it. I have added a check for unexpected events.